axum/src/extract/path/mod.rs

```diff
@@ -641,6 +641,41 @@ mod tests {
         assert_eq!(res.status(), StatusCode::OK);
     }
 
+    #[crate::test]
+    async fn captures_match_empty_inner_segments() {
+        let app = Router::new().route(
+            "/:key/method",
+            get(|Path(param): Path<String>| async move { param.to_string() }),
+        );
+
+        let client = TestClient::new(app);
+
+        let res = client.get("/abc/method").send().await;
+        assert_eq!(res.text().await, "abc");
+
+        let res = client.get("//method").send().await;
+        assert_eq!(res.text().await, "");
+    }
+
+    #[crate::test]
+    async fn captures_match_empty_trailing_segment() {
+        let app = Router::new().route(
+            "/method/:key",
+            get(|Path(param): Path<String>| async move { param.to_string() }),
+        );
+
+        let client = TestClient::new(app);
+
+        let res = client.get("/method/abc").send().await;
+        assert_eq!(res.text().await, "abc");
+
+        let res = client.get("/method/").send().await;
+        assert_eq!(res.text().await, "");
+
+        let res = client.get("/method").send().await;
+        assert_eq!(res.status(), StatusCode::NOT_FOUND);
+    }
```

Comment:
Nice!

Wanna add a similar test for `/method/:key/`? So with a trailing slash.

Reply:
Or do you mean another route handler for that?

Reply:
Yeah that's what I meant. A new test with `/method/:key/`.